app/models/ambassador.rb

```diff
@@ -0,0 +1,41 @@
+class Ambassador < SimpleDelegator
+  def self.all
+    decorate(User.ambassadors)
+  end
+
+  def self.decorate(target)
+    unless target.is_a?(User) || target.respond_to?(:all?) && target.all? { |e| e.is_a?(User) }
+      raise ArgumentError, "Must be a User or collection of Users"
+    end
+
+    Array(target).map { |u| new(u) }
+  end
+
+  def initialize(user)
+    unless user.is_a?(User) && user.ambassador?
+      raise ArgumentError, "Must be an ambassador"
+    end
+    super(user)
+  end
+
+  def percent_complete
+    return 0.0 if ambassador_task_assignments.empty?
+    (completed_tasks_count / tasks_count.to_f).round(2)
+  end
+
+  def progress_count
+    "#{completed_tasks_count}/#{tasks_count}"
+  end
+
+  def completed_tasks_count
+    ambassador_task_assignments.select(&:completed?).count
```

Comment:
Similarly, this isn't like sql and will instantiate all the `ambassador_task_assignments` - if you add a `completed` scope to `ambassador_task_assignment` and do `ambassador_task_assignments.completed.count`, you will be executing only one sql command count